### PR TITLE
Fixes the current create_subscription api call

### PR DESCRIPTION
**Pull request by koetsier**
Previously, when a user attempted to create a subscription multiple times the code would end the existing subscription, recreate the subscription and send an email.

This PR fixes the behaviour of the API so that it returns a no-op success when the specified subscription already exists. The API still replaces an existing subscription if the one specified has a different frequency.

Trello: https://trello.com/c/L9Oxr2Qx/279-double-opt-in-ignore-multiple-attempts-to-subscribe

### CODE REVIEW

File: app/controllers/subscriptions_controller.rb [hunk 1]
```diff
@@ -2,31 +2,36 @@ class SubscriptionsController < ApplicationController
...
-      subscription = Subscription.create!(
-        subscriber: subscriber,
-        subscriber_list: subscriber_list,
-        frequency: frequency,
-        signon_user_uid: current_user.uid,
-        source: existing_subscription ? :frequency_changed : :user_signed_up,
-      )
+      subscription = new_subscription || existing_subscription
+      send_email = create_new_subscription || deactivated_subscriber
+      status = existing_subscription.nil? ? :created : :ok
```
Comment (benthorner) [inline app/controllers/subscriptions_controller.rb]: Should this be `created` if `create_new_subscription`?

File: doc/api.md [hunk 2]
```diff
@@ -178,13 +178,18 @@ new email address.
...
 ```
 
 and it will create a new subscription between the email address and the
 subscriber list. It will respond with a `201 Created` if it's a new
-subscription or a `200 OK` if the subscription already exists.
+subscription or a `200 OK` if the subscription already exists. If a 
+subscription already exists but the frequency is different, the
+current subscription is ended and a new one with the updated frequency
+is created. A confirmation email will be sent if a new subscription is 
+created.
```
Comment (benthorner) [inline doc/api.md]: Minor: could we add a quick note that this is also if the subscriber is re-activated?

File: app/controllers/subscriptions_controller.rb [hunk 3]
```diff
@@ -2,31 +2,36 @@ class SubscriptionsController < ApplicationController
...
 
-      subscription = Subscription.create!(
-        subscriber: subscriber,
-        subscriber_list: subscriber_list,
-        frequency: frequency,
-        signon_user_uid: current_user.uid,
-        source: existing_subscription ? :frequency_changed : :user_signed_up,
-      )
+      subscription = new_subscription || existing_subscription
+      send_email = create_new_subscription || deactivated_subscriber
```
Comment (benthorner) [inline app/controllers/subscriptions_controller.rb]: Minor, but `deactivated_subscriber` is strange to read down here. How about changing the assignment to be

```
if subscriber.deactivated?
   subscriber_was_deactivated = true
   subscriber.activate!
end
```

It's not great, but it does mean the subsequent code is easier to understand.

File: app/controllers/subscriptions_controller.rb [hunk 4]
```diff
@@ -2,31 +2,36 @@ class SubscriptionsController < ApplicationController
...
         subscriber_list: subscriber_list,
       )
 
-      existing_subscription.end(reason: :frequency_changed) if existing_subscription
+      create_new_subscription = existing_subscription.nil? ||
+        (frequency.to_s != existing_subscription.frequency.to_s)
 
-      subscriber.activate! if subscriber.deactivated?
+      if create_new_subscription
+        existing_subscription&.end(reason: :frequency_changed)
```
Comment (benthorner) [inline app/controllers/subscriptions_controller.rb]: Minor: I still find the `frequency_changed` stuff hard to follow, since you have to know about how `create_new_subscription` can be true. I think it might be easier to read this if we had another variable of `frequency_changed` and did the 'ending stuff' in a separate `if` block. How does that sound?

File: doc/api.md (hunk 2 above)
Comment (koetsier) [reply doc/api.md]: 👍

File: app/controllers/subscriptions_controller.rb (hunk 1 above)
Comment (koetsier) [reply app/controllers/subscriptions_controller.rb]: That was not the behaviour of the previous code. I suppose it depends on how you think about it - if you end the previous subscription and create a new one with a different frequency you could see it as modifying the the existing subscription

File: app/controllers/subscriptions_controller.rb (hunk 4 above)
Comment (koetsier) [reply app/controllers/subscriptions_controller.rb]: I tried a few permutations but still can't come up with something better looking than this.

File: app/controllers/subscriptions_controller.rb (hunk 3 above)
Comment (koetsier) [reply app/controllers/subscriptions_controller.rb]: updated it. let me know what you think!
